Source/WebService.swift

```diff
@@ -99,13 +124,24 @@ public struct WebService {
         
         let configuration = resource.configuration
         
-        // Set url
-        guard let url = URL(string: configuration.basePath)?.appendingPathComponent(resource.urlPath) else {
-            throw WebServiceConfigurationError.couldNotBuildUrl
+        let resourceUrl: URL
+        
+        // Set URL
+        if let basePath = configuration.basePath, !basePath.isEmpty {
```

Comment:
comment on what is going on in each case would be useful
like: `//building url for no base path`